collectors/python.d.plugin/mongodb/mongodb.chart.py

```diff
@@ -707,7 +708,7 @@ def _create_connection(self):
         try:
             connection = MongoClient(**conn_vars)
             if self.user and self.password:
-                connection.admin.authenticate(name=self.user, password=self.password)
+                connection[self.authdb].authenticate(name=self.user, password=self.password)
```

Comment:
i see requirements is `python-pymongo package v2.4+.`

it would be very nice if you check that `connection` in py2.4 has `__getitem__` method. It is very likely, but worth to check.

Reply:
Are you referring to PyMongo v2.4 or Python v2.4? The former has __getitem__ even in version 1.0, but is the latter really used anywhere? I recall even old CentOS has older version and I doubt netdata is used in such cases.

Reply:
I guess `getattr()` call instead of array subscript style which calls `__getitem__()` should fit, if Python below 2.4 support is in concern.

Reply:
Done

Reply:
i am referring to pymongo2.4

I am talking about this change
`connection.admin` to `connection[self.authdb]`

ok, i see from the code
```python
    def __getattr__(self, name):
        """Get a database by name.

        Raises :class:`~pymongo.errors.InvalidName` if an invalid
        database name is used.

        :Parameters:
          - `name`: the name of the database to get
        """
        return database.Database(self, name)

    def __getitem__(self, name):
        """Get a database by name.

        Raises :class:`~pymongo.errors.InvalidName` if an invalid
        database name is used.

        :Parameters:
          - `name`: the name of the database to get
        """
        return self.__getattr__(name)
```

so `connection.admin` == `connection[self.authdb]`, i wonder if it is true for pymongo 2.4

Reply:
ok, this code haven't been changed since 2.4